### DISCUSSION TITLE
Trying to retrieve all existing language resources of a global navigation Title in SPO

### DISCUSSION THREAD

**Discussion by rolfm-siemens-com** [pnp/powershell#1386] [Q&A]
Hello Community, 
We have set up global navigation in one our QA tenants and now want to transport the navigation structure to the Productive tenant. Copy process works fine but we saw, that some of the navigation topics have multi lingual titles and I need to transport them also.

Retrieving navigation node with
`$item=Get-PnPNavigationNode -connection $sourceConnection -Id $navItem.Id`
result is
_AudienceIds        : 
Children           : 
CurrentLCID        : 1031
Id                 : 2020
IsDocLib           : True
IsExternal         : True
IsVisible          : True
ListTemplateType   : NoListTemplate
Title              : Länder
TitleResource      : Microsoft.SharePoint.Client.UserResource
Url                : http://linkless.header/
Context            : PnP.Framework.PnPClientContext
Tag                : 
Path               : Microsoft.SharePoint.Client.ObjectPathIdentity
ObjectVersion      : 
ServerObjectIsNull : False
TypedObject        : Microsoft.SharePoint.Client.NavigationNode_

This Navigation node has a title in 
- german 'de-de'  "Länder" 
- english 'en-en'  "Countries"

TitleResouce property shows
_Context            : PnP.Framework.PnPClientContext
Tag                : 
Path               : Microsoft.SharePoint.Client.ObjectPathProperty
ObjectVersion      : 
ServerObjectIsNull : 
TypedObject        : Microsoft.SharePoint.Client.UserResource_

Now I need to retrieve all existing language instances of that title

tried with
`$item.TitleResource.GetResourceEntries()`
-no response
`$item.TitleResource.GetValueForUICulture('de-de')`
-no Response

I assume I need to load / retrieve the content somehow, **but I do not have any idea here**. 
I assume that the process will be the same for all multilingual properties.

Every help is much appreciated.
Environment is pnp.powershell V1.8.0

Best Regards, 
Rolf.

**Comment by agtenr** [comment]
I'm encountering the same issue. 
@erwinvanhunen, Can you shed some light on how to work with multilingual navigation node? 

Thanks in advance!

**Comment by LeonArmston** [comment]
Hi @rolfm-siemens-com & @agtenr 

I think you are misingg that after the methods you call for GetResourceEntries and GetValueForUICulture you need to run [Invoke-PnPQuery](https://github.com/pnp/powershell/blob/dev/documentation/Invoke-PnPQuery.md) to execute the method.

```powershell
$node = Get-PnPNavigationNode -Id <ID of navigation element>


$germanTitle = $node.TitleResource.GetValueForUICulture('de-de')
Invoke-PnPQuery
Write-Host "German Title Is: $($germanTitle.Value)"

$ResourceEntries = $node.TitleResource.GetResourceEntries()
Invoke-PnPQuery
$ResourceEntries #Resource Entries In This Variable
```

Which results in the output below showing the German translation using GetValueForUICulture and the all the title resource entries (showing English, German and Spanish translation).

![image](https://user-images.githubusercontent.com/12968962/153613158-1b0e8820-f926-42b4-8ecc-a9fab13834c6.png)

Hope this helps

**Reply by agtenr** [reply]
@LeonArmston Thanks for the information. This makes it very clear. 

Do you know if there is a way to create new navigation nodes with multiple languages?

**Reply by LeonArmston** [reply]
Create new Navigation node using [Add-PnPNavigationNode](https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnpnavigationnode?view=sharepoint-ps) then you can add translations by using the method SetValueForUICulture like below:


```powershell

$node = Get-PnPNavigationNode -Id 2058

$node.TitleResource.SetValueForUICulture('de-DE','Meine Werkzeuge')
$node.Update()
Invoke-PnPQuery
```

**Reply by agtenr** [reply]
@LeonArmston, you are a lifesaver. Thanks for this information! Might be useful to add this to the documentation!
